codec.go

```diff
@@ -136,6 +156,15 @@ func (c *protoJSONCodec) Marshal(message any) ([]byte, error) {
 	return options.Marshal(protoMessage)
 }
 
+func (c *protoJSONCodec) MarshalAppend(dst []byte, message any) ([]byte, error) {
+	protoMessage, ok := message.(proto.Message)
+	if !ok {
+		return nil, errNotProto(message)
+	}
+	var options protojson.MarshalOptions
+	return options.MarshalAppend(dst, protoMessage)
```

Comment:
Nit: Any reason not to use `protojson.MarshalOptions{}.MarshalAppend` like above for consistency?